docs/how-to/h-deploy-aws.md

```diff
@@ -0,0 +1,210 @@
+Located Juju agent version 3.5.4-ubuntu-amd64 at https://juju-dist-aws.s3.amazonaws.com/agents/agent/3.5.4/juju-3.5.4-linux-amd64.tgz
+Launching controller instance(s) on aws/us-east-1...
+ - i-0f4615983d113166d (arch=amd64 mem=8G cores=2)           
+Installing Juju agent on bootstrap instance
+Waiting for address
+Attempting to connect to 54.226.221.6:22
+Attempting to connect to 172.31.20.34:22
+Connected to 54.226.221.6
+Running machine configuration script...
+Bootstrap agent now started
+Contacting Juju controller at 54.226.221.6 to verify accessibility...
+
+Bootstrap complete, controller "aws-us-east-1" is now available
+Controller machines are in the "controller" model
+
+Now you can run
+	juju add-model <model-name>
+to create a new model to deploy workloads.
+```
+[/details]
+
+## Deploy charms
+
+Create a new Juju model, if needed:
+
+```shell
+juju add-model <MODEL_NAME>
+```
+
+[note type="caution"]
+(Optional) Increase the debug level if you are troubleshooting charms:
+```shell
+juju model-config logging-config='<root>=INFO;unit=DEBUG'
+```
+[/note]
+
+Then, Charmed Kafka can be deployed as usual. However, note that the smallest instance types on AWS may not have enough resources for hosting 
+a Kafka broker. We recommend selecting an instance type that provides at the very least 8 GB of RAM and 4 cores.
+For more guidance on production environment sizing, see the [Requirements page](/t/charmed-kafka-reference-requirements/10563).
+You can find more information about the available instance types in the [AWS documentation](https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#Instances:instanceState=running), 
+ensuring that you select the correct AWS.
+
+Deploy and integrate Kafka and ZooKeeper, for example:
+
+```shell
+juju deploy zookeeper -n3 --channel 3/stable
+juju deploy kafka -n3 --constraints "instance-type=m7i.xlarge" --channel 3/stable
+juju integrate kafka zookeeper
+```
+
+We also recommend to deploy a [Data Integrator](https://charmhub.io/data-integrator) for creating an admin user to manage the content of the Kafka cluster:
+
+```shell
+juju deploy data-integrator admin \
+  --config extra-user-roles=admin \
+  --config topic-name=admin-topic
+```
+
+And integrate it with the Kafka application:
+
+```shell
+juju integrate kafka admin
+```
+
+For more information on Data Integrator and how to use it, please refer to the [how-to manage applications](/t/charmed-kafka-how-to-manage-app/10285) guide.
+
+## Clean up
+
+[note type="caution"]
+Always clean AWS resources that are no longer necessary! Abandoned resources are tricky to detect and they can become expensive over time.
+[/note]
+
+To list all controllers use the `juju controllers` command.
+
+To destroy the Juju controller and remove AWS instance (**Warning**: all your data will be permanently deleted):
+
+```shell
+juju destroy-controller <CONTROLLER_NAME> --destroy-all-models --destroy-storage --force
```

Comment:
this command is known to hang indefinitely unless a timeout is specified. By default, there is no timeout. Specify the command with a timeout of 5 minutes.

Reply:
Not really, I generally use this command with no issues. Sometimes it may hangs when it tries to clear some resources (especially on K8s) when some resources hangs on deletion. But it is still good that it keeps on trying. While the process runs, I also (in a different shell) delete resource manually to get the process unstuck and resume. I would not nudge away to use timeout (why wouldn't we go with the default provided by juju?), user forgetting (or thinking everything is fine) then having resources hanging